pkg/network/bridge/lister.go

```diff
@@ -15,26 +20,60 @@ const (
 	maxBridgeNameLength = 12
 )
 
+func GetBridgeList(configMap *v1.ConfigMap) dpm.PluginNameList {
+	var plugins = make(dpm.PluginNameList, 0)
+
+	bridges := strings.Split(configMap.Data["bridges"], ",")
+
+	for _, bridgeName := range bridges {
+		if len(bridgeName) > maxBridgeNameLength {
+			glog.Fatalf("Bridge name (%s) cannot be longer than %d characters", bridgeName, maxBridgeNameLength)
+		}
+		plugins = append(plugins, bridgeName)
+	}
+
+	return plugins
+}
+
 type BridgeLister struct{}
 
 func (bl BridgeLister) GetResourceNamespace() string {
 	return resourceNamespace
 }
 
 func (bl BridgeLister) Discover(pluginListCh chan dpm.PluginNameList) {
-	var plugins = make(dpm.PluginNameList, 0)
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		panic(err.Error())
+	}
 
-	bridgesListRaw := os.Getenv(BridgesListEnvironmentVariable)
-	bridges := strings.Split(bridgesListRaw, ",")
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		panic(err.Error())
+	}
 
-	for _, bridgeName := range bridges {
-		if len(bridgeName) > maxBridgeNameLength {
-			glog.Fatalf("Bridge name (%s) cannot be longer than %d characters", bridgeName, maxBridgeNameLength)
-		}
-		plugins = append(plugins, bridgeName)
+	configmap, err := clientset.CoreV1().ConfigMaps("kube-system").Get("device-plugin-network-bridge", metav1.GetOptions{})
+	if err != nil {
+		panic(err.Error())
 	}
 
-	pluginListCh <- plugins
+	pluginListCh <- GetBridgeList(configmap)
```

Comment:
The informer will only watch for updates so if the plugin pod got restarted no bridges would be discovered until the config map is updated again. Line 60 passes the existing bridges in the config map to the channel for the initial discovery.

Reply:
The current solution is raceful, the map may change between this line and startup of the informer.

If you add AddFunc callback to the informer, you would get the initial map (maybe you need to set an extra argument or I am totally wrong).

Also if someone changes the map by removing it and creating again, this would not work either.

Reply:
@phoracek You are correct sir. I have pushed that change. Let me know if there's anything else you spot.